cmd/minikube/cmd/config/scan.go

```diff
@@ -0,0 +1,130 @@
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package config
+
+import (
+	"encoding/json"
+	"os"
+	"os/exec"
+
+	"github.com/spf13/cobra"
+	"gopkg.in/yaml.v2"
+	"k8s.io/klog/v2"
+	"k8s.io/minikube/pkg/minikube/download"
+	"k8s.io/minikube/pkg/minikube/exit"
+	"k8s.io/minikube/pkg/minikube/reason"
+	"k8s.io/minikube/pkg/util"
+)
+
+var addonsScan = &cobra.Command{
+	Use:    "scan",
+	Short:  "Scans all minikube addon images for security vulnerabilities",
+	Long:   "Scans all minikube addon images for security vulnerabilities",
+	Hidden: true,
+	Run: func(cmd *cobra.Command, args []string) {
+		addonMap, err := download.AddonList()
+		if err != nil {
+			exit.Error(reason.InternalAddonScan, "getting addon list", err)
+		}
+
+		oldStatusMap, err := download.AddonStatus()
+		// If we can't find the old status file, we can just proceed
+		if err != nil {
+			klog.Warningf("failed to get old addon status list: %v", err)
+		}
+
+		addonStatus := make(map[string]util.AddonStatus)
+		for a, i := range addonMap {
+			klog.Infof("scanning addon %s", a)
+			images := i.(map[interface{}]interface{})
+			status := util.AddonStatus{Enabled: true, Manual: false, ManualReason: "", Images: []util.ImageStatus{}}
+			if s, ok := oldStatusMap[a]; ok {
+				if !s.Enabled && s.Manual {
+					// Addon was manually disabled, respect that
+					status = s
+					addonStatus[a] = status
+					continue
+				}
+			}
+			for _, image := range images {
+				klog.Infof("scanning image %s for addon %s", image, a)
+				imageStatus := util.ImageStatus{Image: image.(string), CVEs: []util.CVE{}}
+				snyk := exec.Command("snyk", "container", "test", image.(string), "--json", "--severity-threshold=high")
+				out, err := snyk.Output()
+				if err == nil {
+					klog.Infof("no vulnerabilities found for %s", image)
+					continue
+				}
+				outmap := make(map[string]interface{})
+				err = json.Unmarshal(out, &outmap)
+				if err != nil {
+					klog.Errorf("error unmarshalling json for %s: %v", image, err)
+				}
+				// The vulnerabilities entry won't show up if there was an error from snyk
+				if vulnz, ok := outmap["vulnerabilities"].([]interface{}); ok {
+					for _, v := range vulnz {
+						vuln := v.(map[string]interface{})
+						logCVE := true
+						for _, c := range imageStatus.CVEs {
+							if c.Name == vuln["title"].(string) {
+								klog.Infof("already logged CVE %s for image %s, skipping", c.Name, image)
+								logCVE = false
+								break
+							}
+						}
+						if !logCVE {
+							continue
+						}
+						klog.Infof("CVE for %s: %s, %s, %s, %v", image, vuln["title"], vuln["packageName"], vuln["severity"], vuln["nearestFixedInVersion"])
+						status.Enabled = false
+						updatedVersion := ""
+						if uv, ok := vuln["nearestFixedInVersion"].(string); ok {
```

Comment:
Optional, we can declare without initialization here.

var updatedVersion string

Reply:
i specifically want this to be an empty string in case it doesn't get set, so we can still use it in the CVE object below